sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -1319,3 +1332,40 @@ async def read_feed_ranges(
 
         return [FeedRangeEpk(Range.PartitionKeyRangeToRange(partitionKeyRange))
                 for partitionKeyRange in partition_key_ranges]
+
+    async def get_updated_session_token(self,
+                                  feed_ranges_to_session_tokens: List[Tuple[str, FeedRange]],
+                                  target_feed_range: FeedRange
+                                  ) -> "Session Token":
+        """Gets the the most up to date session token from the list of session token and feed range tuples
+        for a specific target feed range. The feed range can be obtained from a response from any crud operation.
+        This should only be used if maintaining own session token or else the sdk will keep track of
+        session token.
+        :param feed_ranges_to_session_tokens: List of partition key and session token tuples.
+        :type feed_ranges_to_session_tokens: List[Tuple(str, FeedRange)]
+        :param target_feed_range: feed range to get most up to date session token.
+        :type target_feed_range: FeedRange
+        :returns: a session token
+        :rtype: str
+        """
+        return get_updated_session_token(feed_ranges_to_session_tokens, target_feed_range)
+
+    async def feed_range_from_partition_key(self, partition_key: PartitionKeyType) -> FeedRange:
+        """Gets the feed range for a given partition key.
+        :param partition_key: partition key to get feed range.
+        :type partition_key: PartitionKey
+        :returns: a feed range
+        :rtype: FeedRange
+        """
+        return FeedRangeEpk(await self._get_epk_range_for_partition_key(partition_key))
+
+    async def is_feed_range_subset(self, parent_feed_range: FeedRange, child_feed_range: FeedRange) -> bool:
```

Comment:
It looks like this doesn't need to be on the client at all? Can this not simply be a utility on the `FeedRange` object itself?
As a Python developer, it would be cool if I could just do something like:
`if child_feed_range in parent_feed_range:`
Or failing that, even something like:
`if parent_feed_range.contains(child_feed_range):`
`if child_feed_range.is_subset(parent_feed_range):`

Reply:
Added container information to the feed ranges and a check to make sure the feed ranges inputted correspond to the correct container. Feed ranges are scoped to a container.

Reply:
Moved to how I originally had it with no checks for if they're in the same container. We still want to to keep it in the container because we might need the container context for future feed range implementations. Also I would like to make those methods in the async container class async as well for the same reason.